woocommerce/woocommerce
# Add error state to variation list

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds an error state when we can't load variations for any reason.

**Design**
AkVNGImLgSqCObTQ3idVn7-fi-1641_222148

Closes #40345.

**Acceptance criteria**

- [ ] Shown when the variations have been created elsewhere (e.g. in the legacy UX) or contain unexpected/incompatible content, or there was an error while creating them.
- [ ] Clicking “Try again” triggers the loading state.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a WooCommerce store with this branch and enable the product editor using **WooCommerce > Settings > Advanced > Features**
2. Go to **Products > Add New > Variations**
3. Press `Add variation options`
4. Add an attribute and a few values (at least two).
5. Now delete the variations that have been created
6. Verify the empty state for the table is visible. Try the `Generate from options` button

![Screenshot 2023-11-17 at 11 22 13](https://github.com/woocommerce/woocommerce/assets/1314156/dadb0e4f-c8a4-41b1-a126-1dea7a4dda50)

7. Don't leave the page
8. Now break the variations REST API (to accomplish that, you can add a random text in [this file](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php))
9. Press the `Edit` button, present in the same line as the attribute you just created in the `Variation options` section. 
10. Remove one of the values you created and press `Update`. That will create the variations again
11. The error state should be visible

![Screenshot 2023-11-17 at 12 42 43](https://github.com/woocommerce/woocommerce/assets/1314156/1b9454e5-94c1-43e0-b10f-b1b15b390bac)

12. Now remove the breaking change added in step 8 and press `Try again`
13. The variations should be created/removed. The error state should disappear.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
